pages/activity/[name]/manage/evaluation.tsx

```diff
@@ -0,0 +1,165 @@
+  InputGroup,
+  ListGroup,
+  Row,
+} from 'react-bootstrap';
+import { formToJSON } from 'web-utility';
+
+import { ActivityManageFrame } from '../../../../components/Activity/ActivityManageFrame';
+import { TeamAwardList } from '../../../../components/Team/TeamAwardList';
+import activityStore from '../../../../models/Activity';
+import { Award, AwardAssignment } from '../../../../models/Award';
+import { Team } from '../../../../models/Team';
+import { withRoute } from '../../../api/core';
+
+interface EvaluationPageProps {
+  activity: string;
+  path: string;
+  awardList: Award[];
+}
+
+export const getServerSideProps = withRoute<{ name: string }>();
+
+@observer
+class EvaluationPage extends PureComponent<
+  InferGetServerSidePropsType<typeof getServerSideProps>
+> {
+  store = activityStore.teamOf(this.props.route.params!.name);
+  awardStore = activityStore.awardOf(this.props.route.params!.name);
+  form = createRef<HTMLFormElement>();
+
+  componentDidMount() {
+    this.awardStore.getAll();
+  }
+
+  handleAssign = () => {};
+  handleReset = () => {
+    const { store } = this,
+      form = this.form.current;
+    form?.reset();
+    store.clearCurrent();
+  };
+
+  handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const { awardStore, store } = this,
+      form = this.form.current;
+
+    if (!store || !form) return;
+
+    const data = formToJSON<NewData<AwardAssignment>>(form);
+
+    await awardStore.getOne(data.awardId!);
+
+    const assignmentStore = awardStore.assignmentOf(data.awardId!),
+      assigneeId = store.currentOne.id;
+
+    await assignmentStore.updateOne({ assigneeId });
+    await store.refreshList();
+
+    store.clearCurrent();
+    form.reset();
+  };
+
+  onSearch = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const { search } = formToJSON<{ search: string }>(event.currentTarget);
+
+    this.store.clear();
+    return this.store.getList({ search });
+  };
+
+  renderForm = () => {
+    const { allItems } = this.awardStore,
+      { id: awardTeamId, displayName: awardTeamName } = this.store.currentOne;
+
+    return (
+      <Form
+        className="p-3 text-nowrap border"
+        ref={this.form}
+        onReset={this.handleReset}
+        onSubmit={this.handleSubmit}
+      >
+        <h2>{awardTeamId ? `授予${awardTeamName}` : '全部奖项'}</h2>
+        {allItems
+          .filter(({ target }) => target === 'team')
+          .map(({ id, name, quantity }) =>
+            awardTeamId ? (
+              <Form.Check
+                type="radio"
+                className="mx-2 my-3"
+                key={id}
+                label={name}
+                name="awardId"
+                value={id}
+              />
+            ) : (
+              <li key={id} className="list-unstyled mx-2 my-3">
+                {name}
+              </li>
+            ),
+          )}
```

Comment:
```suggestion
      <ul className="list-unstyled">
        {allItems.map(({ target, id, name, quantity }) =>
            (target === 'team') && 
              <li key={id} className="mx-2 my-3">{
                  awardTeamId ? (
              <Form.Check
                type="radio"
                key={id}
                label={name}
                name="awardId"
                value={id}
              />
            ) : name}
              </li>,
          )}
        </ul>
```
React 会自动忽略 JSX 中数组元素中的假值